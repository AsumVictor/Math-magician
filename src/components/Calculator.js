import React, { useState } from 'react';
import Screen from './Screen';
import Button from './Button';
import calculate from '../logic/Calculator';

function Calculator() {
  const [calculation, setCalculation] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const makeCalculations = (e) => {
    const buttonName = e.currentTarget.textContent;
    const newState = calculate(calculation, buttonName);
    setCalculation(newState);
  };

  const { total, next, operation } = calculation;

  return (
    <div className="w-[9cm] md:w-[12cm] h-[9cm] md:h-[12cm] border m-10 grid grid-cols-4 grid-rows-6 ">
      <Screen total={total} next={next} operation={operation} />
      <Button handleClick={(e) => makeCalculations(e)} value="AC" />
      <Button handleClick={(e) => makeCalculations(e)} value="+/-" />
      <Button handleClick={(e) => makeCalculations(e)} value="%" />
      <Button handleClick={(e) => makeCalculations(e)} value="รท" classextend="bg-orange-500" />
      <Button handleClick={(e) => makeCalculations(e)} value="7" />
      <Button handleClick={(e) => makeCalculations(e)} value="8" />
      <Button handleClick={(e) => makeCalculations(e)} value="9" />
      <Button handleClick={(e) => makeCalculations(e)} value="x" classextend="bg-orange-500" />
      <Button handleClick={(e) => makeCalculations(e)} value="4" />
      <Button handleClick={(e) => makeCalculations(e)} value="5" />
      <Button handleClick={(e) => makeCalculations(e)} value="6" />
      <Button handleClick={(e) => makeCalculations(e)} value="-" classextend="bg-orange-500" />
      <Button handleClick={(e) => makeCalculations(e)} value="1" />
      <Button handleClick={(e) => makeCalculations(e)} value="2" />
      <Button handleClick={(e) => makeCalculations(e)} value="3" />
      <Button handleClick={(e) => makeCalculations(e)} value="+" classextend="bg-orange-500" />
      <Button handleClick={(e) => makeCalculations(e)} value="0" classextend="col-span-2 bg-slate-200 text-black" />
      <Button handleClick={(e) => makeCalculations(e)} value="." />
      <Button handleClick={(e) => makeCalculations(e)} value="=" classextend="bg-orange-500" />
    </div>
  );
}

export default Calculator;
