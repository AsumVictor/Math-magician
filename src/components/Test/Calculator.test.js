// Calculator.test.js
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Calculator from '../Calculator';

test('renders Calculator component', () => {
  render(<Calculator />);
});

test('performs addition correctly', () => {
  const { getByText } = render(<Calculator />);
  const button2 = getByText('2');
  const buttonPlus = getByText('+');
  const buttonEquals = getByText('=');

  fireEvent.click(button2);
  fireEvent.click(buttonPlus);
  fireEvent.click(button2);
  fireEvent.click(buttonEquals);
});

test('clears the calculation on AC button click', () => {
  const { getByText } = render(<Calculator />);
  const button2 = getByText('2');
  const buttonPlus = getByText('+');
  const buttonAC = getByText('AC');

  fireEvent.click(button2);
  fireEvent.click(buttonPlus);
  fireEvent.click(button2);
  fireEvent.click(buttonAC);
});
