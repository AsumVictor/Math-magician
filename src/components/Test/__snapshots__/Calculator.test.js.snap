// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `
<div>
  <div
    class="w-[9cm] md:w-[12cm] h-[9cm] md:h-[12cm] border m-10 grid grid-cols-4 grid-rows-6 "
  >
    <div
      class="col-span-4 bg-gray-700 px-2 flex justify-end items-center text-white"
    >
      <span />
      <span />
      <span>
        0
      </span>
    </div>
    <button
      class="border flex items-center justify-center border-white font-bold text- rounded-md bg-slate-200 text-black "
      type="button"
    >
      AC
    </button>
    <button
      class="border flex items-center justify-center border-white font-bold text- rounded-md bg-slate-200 text-black "
      type="button"
    >
      +/-
    </button>
    <button
      class="border flex items-center justify-center border-white font-bold text- rounded-md bg-slate-200 text-black "
      type="button"
    >
      %
    </button>
    <button
      class="border flex items-center justify-center border-white font-bold text- rounded-md bg-orange-500 "
      type="button"
    >
      รท
    </button>
    <button
      class="border flex items-center justify-center border-white font-bold text- rounded-md bg-slate-200 text-black "
      type="button"
    >
      7
    </button>
    <button
      class="border flex items-center justify-center border-white font-bold text- rounded-md bg-slate-200 text-black "
      type="button"
    >
      8
    </button>
    <button
      class="border flex items-center justify-center border-white font-bold text- rounded-md bg-slate-200 text-black "
      type="button"
    >
      9
    </button>
    <button
      class="border flex items-center justify-center border-white font-bold text- rounded-md bg-orange-500 "
      type="button"
    >
      x
    </button>
    <button
      class="border flex items-center justify-center border-white font-bold text- rounded-md bg-slate-200 text-black "
      type="button"
    >
      4
    </button>
    <button
      class="border flex items-center justify-center border-white font-bold text- rounded-md bg-slate-200 text-black "
      type="button"
    >
      5
    </button>
    <button
      class="border flex items-center justify-center border-white font-bold text- rounded-md bg-slate-200 text-black "
      type="button"
    >
      6
    </button>
    <button
      class="border flex items-center justify-center border-white font-bold text- rounded-md bg-orange-500 "
      type="button"
    >
      -
    </button>
    <button
      class="border flex items-center justify-center border-white font-bold text- rounded-md bg-slate-200 text-black "
      type="button"
    >
      1
    </button>
    <button
      class="border flex items-center justify-center border-white font-bold text- rounded-md bg-slate-200 text-black "
      type="button"
    >
      2
    </button>
    <button
      class="border flex items-center justify-center border-white font-bold text- rounded-md bg-slate-200 text-black "
      type="button"
    >
      3
    </button>
    <button
      class="border flex items-center justify-center border-white font-bold text- rounded-md bg-orange-500 "
      type="button"
    >
      +
    </button>
    <button
      class="border flex items-center justify-center border-white font-bold text- rounded-md col-span-2 bg-slate-200 text-black "
      type="button"
    >
      0
    </button>
    <button
      class="border flex items-center justify-center border-white font-bold text- rounded-md bg-slate-200 text-black "
      type="button"
    >
      .
    </button>
    <button
      class="border flex items-center justify-center border-white font-bold text- rounded-md bg-orange-500 "
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
